### DISCUSSION TITLE
Single item edit with nested field arrays

### DISCUSSION THREAD

**Discussion by raywong-fb** [react-hook-form/react-hook-form#6065] [Q&A] [answered]
I'm using and loving RHF. I've run across an issue where I have a list of items (let's say in the left column), then when you click on any item it will display an edit component for the selected item in the right column. The issue is when I switch between items, **_nested array values_** don't seem to update the data correctly.

I think I understand that because a new edit component isn't created for each parent item (I'm just passing the selected index down) the nested `useFieldArray` in the edit component still references the first parent item. If that's the case, I'm not exactly sure how to make sure a new `useFieldArray` is created for each item that gets selected.

Here's a codesandbox replicating the issue:
https://codesandbox.io/s/dreamy-brattain-r74lx

Any help is much appreciated.

**Comment by bluebill1049** [comment]
can you try this version? react-hook-form@7.12.1-next.0 see if it solve the problem?

**Reply by raywong-fb** [reply]
Unfortunately, the new version did not seem to fix this issue

**Comment by raywong-fb** [comment, marked as answer]
@bluebill1049 I'm not sure if my original code still points to a bug, but I was able work around the issue w/ the answer given [here](https://stackoverflow.com/a/68502762/643274)

**Comment by raywong-fb** [comment]
@bluebill1049 somewhat related to my original scenario, while it appears the values are correctly updated, when I click on more than one of the "Parent" items, and then add a child to the list, I'm noticing `undefined` array items are being added. (click submit to see a log of the values) Thoughts?

https://codesandbox.io/s/cold-violet-y4i3r?file=/src/App.js

**Reply by bluebill1049** [reply]
can you share some steps? also maybe try the latest `next` version 7.13.0-next.0

**Reply by raywong-fb** [reply]
You can duplicate by following the steps:

1. Click on Parent Item 1
2. Click on Parent Item 2
3. Add a new child item to Parent 2
4. Look at the console logs, notice that after adding a new child item, there is an `undefined` array item

**Reply by bluebill1049** [reply]
did you try the next version?

**Reply by raywong-fb** [reply]
the next version appears to fix the `undefined` issue, thanks!

however, it does not seem to fix the original issue, where using the `fields` from the nested field array does not seem to re-render w/ the correct data.
